docs/developer/extending/apps/developing-with-tunnels.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: Tunneling Apps
+---
+
+## What is a tunnel?
+
+The Saleor extendability model relies on the concept of Apps. If you want to read more about them, head to [the Apps page](developer/extending/apps/key-concepts.mdx).
+
+As you want Apps to react to events in your store, you reach for webhooks. However, webhooks pose a unique challenge: they can only communicate with public clients. If so, how are you supposed to develop your app on `localhost`?
+
+This is where tunnels come into play. **A tunnel allows you to create a proxy that exposes a local environment to a public URL.**
+
+```mermaid
+flowchart LR
+    a[Dev server]
+    t[Tunnel]
+    c[Saleor Cloud]
+    e[External services]
+
+    subgraph Local environment
+        a
+    end
+
+    subgraph Internet
+        t
+        c
+        e
+    end
+
+    a<-.-> t
+    t<-->c
+    t<-->e
+```
+
+---
+
+Let's say you are building a [Next.js](https://nextjs.org/) application that leverages a webhook. You boot it up with the `next dev` command, which renders it on `http://localhost:3000`. Since this address is not HTTPS-compliant, the app cannot access the webhook response. We have a perfect use case for a tunnel on our hands.
```

Comment:
```suggestion
Let's say you are building a [Next.js](https://nextjs.org/) application that leverages a webhook. You boot it up with the `next dev` command, which renders it on `http://localhost:3000`. Since this address is only reachable from your machine and is not HTTPS-compliant, the webhook cannot reach the app. We have a perfect use case for a tunnel on our hands.
```

Events (webhooks) reach for the app, not the other way around. 
The app can reach cloud API (since it's exposed to the internet).